sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2813,13 +2813,41 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
       ctx: CreateOrReplaceTableColTypeContext): StructField = withOrigin(ctx) {
     import ctx._
 
+    // Check that no duplicates exist among any CREATE TABLE column options specified.
+    var isNotNull: Option[Boolean] = None
```

Comment:
seems this can be a boolean, starting with false, and set to true if NOT NULL is present.

Reply:
That's a good point. We can just name this `nullable` and set it to initially `true`, and change it to `false` if the SQL syntax includes `NOT NULL`. Updated accordingly.